uber/config.py

```diff
@@ -13,15 +13,9 @@ def mixin(cls, klass):
     def include_plugin_config(self, plugin_config):
         """Plugins call this method to merge their own config into the global c object."""
 
-        for attr, val in plugin_config.items():
-            if not isinstance(val, dict):
-                setattr(self, attr.upper(), val)
-
-        if 'enums' in plugin_config:
-            self.make_enums(plugin_config['enums'])
-
-        if 'dates' in plugin_config:
-            self.make_dates(plugin_config['dates'])
```

Comment:
Using ``.merge()`` here will merge in literally everything.  Previously the only things that got merged in were:
- top-level single config items
- ``[enums]``
- ``[dates]``

Now we're merging in other stuff like
- badge and table prices
- badge ranges
- age groups
- department head and volunteer checklists
- django and cherrypy config

I originally avoided doing this, because I figured it was better to keep plugin config limited instead of just saying "anything you put in this config will literally override everything in the main config".  On balance, maybe it's just simpler/better this way, especially since plugins are already able to do those things if they really want to regardless. 

With that being said, there's still a subtle-but-serious bug in this implementation which requires too much detail to explain until I get home from work tonight, so hold off on merging this.